cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/to/AbstractRESTSearchResponse.java

```diff
@@ -0,0 +1,111 @@
+   */
+  public List<URL> getDownloadURLs() throws MalformedURLException;
+
+  /**
+   * Gets the json response
+   *
+   * @param repositoryUrl URL of the repository
+   * @param groupId to search for
+   * @return String of json response
+   * @throws IOException if the request did not return status 200
+   */
+  public String getJsonResponse(String repositoryUrl, String groupId) throws IOException;
+
+  /**
+   * Gets the download links by given repository type
+   *
+   * @param repositoryType String of the type of the repository e.g. maven, jfrog, nexus
+   * @param groupId the groupId to search for
+   *
+   * @return List of download links
+   * @throws IOException if an error occurred
+   *
+   */
+  public static List<URL> getArtifactDownloadLinks(String repositoryType, String groupId) throws IOException {
+
+    List<URL> downloadLinks = new ArrayList<>();
+    ObjectMapper mapper = new ObjectMapper();
+    String jsonResponse = "";
+
+    if (repositoryType.equals("maven")) {
+      MavenSearchResponse response = new MavenSearchResponse();
+      String mavenRepositoryURL = "https://search.maven.org";
+      jsonResponse = response.getJsonResponse(mavenRepositoryURL, groupId);
+      response = mapper.readValue(jsonResponse, MavenSearchResponse.class);
+      downloadLinks = response.getDownloadURLs();
+      return downloadLinks;
+    }
+
+    if (repositoryType.equals("jfrog")) {
+      JfrogSearchResponse response = new JfrogSearchResponse();
+      String jfrogRepositoryURL = "";
+      jsonResponse = response.getJsonResponse(jfrogRepositoryURL, groupId);
+      response = mapper.readValue(jsonResponse, JfrogSearchResponse.class);
+      downloadLinks = response.getDownloadURLs();
+      return downloadLinks;
+    }
+
+    if (repositoryType.equals("nexus")) {
+      NexusSearchResponse response = new NexusSearchResponse();
+      String nexusRepositoryURL = "";
+      jsonResponse = response.getJsonResponse(nexusRepositoryURL, groupId);
+      response = mapper.readValue(jsonResponse, NexusSearchResponse.class);
+      downloadLinks = response.getDownloadURLs();
+      return downloadLinks;
+    }
+
+    return null;
+  }
+
+  /**
+   * Gets a json response by given REST API target link
+   *
+   * @param targetLink link to get response from
+   * @return String of json response
+   * @throws IOException if the returned status code was not 200 OK
+   */
+  public static String getJsonResponseStringByTargetLink(String targetLink) throws IOException {
+
+    Client client = ClientBuilder.newClient();
+    WebTarget target = client.target(targetLink);
+    Invocation.Builder request = target.request(MediaType.APPLICATION_JSON);
+    Response response = request.get();
+    int status = response.getStatus();
+    String jsonResponse = "";
+    if (status == 200) {
+      jsonResponse = response.readEntity(String.class);
+    } else {
+      throw new IOException(String.valueOf(status));
```

Comment:
why not HTTPException? or WebApplicationException?

Reply:
Added a new RESTSearchResponseException instead.